### PR TITLE
Automated CipherNode deployment 

### PR DESCRIPTION
**Pull request by hmzakhalid**
## Release Notes

- **New Features**
	- Introduced a new GitHub Actions workflow for building and deploying the Ciphernode application.
	- Added new services (`cn1`, `cn2`, `cn3`, `aggregator`) and a network (`cn-network`) in the development Docker Compose configuration.
	- Enhanced configuration management with new parameters (`quic_port`, `enable_mdns`) across various configuration files.

- **Bug Fixes**
	- Improved error handling and logging in event extraction and processing logic.

- **Documentation**
	- Updated configuration files to reflect new parameters for better flexibility.

- **Chores**
	- Removed deprecated environment variables and adjusted Docker Compose service configurations.

### CODE REVIEW

File: packages/ciphernode/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
+FROM node:20 AS evm-builder
```
Comment (ryardley) [inline packages/ciphernode/Dockerfile:1]: LTS is 22 now I believe - should we keep this current or is there a reason to stick with an older version

File: packages/ciphernode/ciphernode-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+PASSWORD=$(jq -r '.password' "$SECRETS_FILE")
+
+if [ -z "$PRIVATE_KEY" ] || [ -z "$PASSWORD" ]; then
+    echo "Error: Missing 'private_key' or 'password' in secrets file!"
+    exit 1
+fi
+
+# Set password and private key
+echo "Setting password"
+enclave password create --config "$CONFIG_FILE" --password "$PASSWORD"
```
Comment (ryardley) [inline packages/ciphernode/ciphernode-entrypoint.sh:32]: Cant we just mount the password file? Or is this just about automating the private key as well as the password?

File: .github/workflows/ecs-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+name: Build and Deploy Ciphernode to ECS
+
+on:
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (hmzakhalid) [reply .github/workflows/ecs-deployment.yml:6]: I am in favor of either, depending on what suits our workflow best. If we decide to make the `main` branch the deployment branch, we’ll need to make sure that only fully tested and production-ready features are merged into it. It'll simplify the branching strategy, as everything ready for deployment will reside in one place. But we'll need strict checks in place to maintain the integrity of the branch and avoid any incomplete or experimental code being merged into it.

if we opt for a dedicated `devnet` or `release` deployment branch then we gain more control over the deployment process and we can essentially stage code for deployment, getting an extra layer of verification before it goes live. we can selectively deploy specific versions and make sure only the correct and approved code reaches production. This strategy is also useful for multi network environments as it makes managing releases easier.

[the remark this answers is not in the document]
Comment (ryardley) [reply .github/workflows/ecs-deployment.yml:6]: > if we opt for a dedicated devnet or release deployment branch then we gain more control over the deployment process and we can essentially stage code for deployment, getting an extra layer of verification before it goes live. we can selectively deploy specific versions and make sure only the correct and approved code reaches production. This strategy is also useful for multi network environments as it makes managing releases easier.

Yeah I think I am in favour of this. @auryn-macmillan might have some stuff to say here?

**Review by ryardley — APPROVED** [review]
🤖

**Pull request merged**